controllers/disruption_controller.go

```diff
@@ -364,12 +361,15 @@ func (r *DisruptionReconciler) startInjection(instance *chaosv1beta1.Disruption)
 				r.Recorder.Event(instance, corev1.EventTypeNormal, "Created", fmt.Sprintf("Created disruption injection pod for \"%s\"", instance.Name))
 				r.recordEventOnTarget(instance, target, corev1.EventTypeWarning, "Disrupted", fmt.Sprintf("Pod %s from disruption %s targeted this resourcer for injection", chaosPod.Name, instance.Name))
 				r.handleMetricSinkError(r.MetricsSink.MetricPodsCreated(target, instance.Name, instance.Namespace, true))
-			} else {
+			case 1:
+				r.log.Infow("an injection pod is already existing for the selected target", "target", target, "chaosPod", found[0].Name)
+			default:
 				var chaosPodNames []string
 				for _, pod := range found {
 					chaosPodNames = append(chaosPodNames, pod.Name)
 				}
-				r.log.Infow("an injection pod is already existing for the selected target", "target", target, "chaosPods", strings.Join(chaosPodNames, ","))
+
+				r.log.Errorw("multiple injection pods for one target found", "target", target, "chaosPods", strings.Join(chaosPodNames, ","), "chaosPodLabels", chaosPod.Labels)
```

Comment:
I don't know if we already had that case (which is definitely a bug/a race condition?). Would it be worth having a metric to track that at some point?

Reply:
I'm not sure I've ever seen this case, I'm fine either way w/ a metric?